pilot/pkg/model/jwks_resolver.go

```diff
@@ -351,7 +352,12 @@ func (r *JwksResolver) getRemoteContentWithRetry(uri string, retry int) ([]byte,
 		}
 
 		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-			return nil, fmt.Errorf("status %d, %s", resp.StatusCode, string(body))
+			message := strconv.Quote(string(body))
```

Comment:
should we also use https://pkg.go.dev/io#LimitedReader on L349? Since we still read the entire response into memory right now....

Reply:
The closed PR https://github.com/istio/istio/pull/35723 still includes those changes.
I did a couple of tests and I could not get my istiod to crash, because of the 5 second timeout. I simply didn't manage to download enough json within 5 seconds to OOM istiod. 
Since there wasn't a consensus on how high we should limit it, I removed it.